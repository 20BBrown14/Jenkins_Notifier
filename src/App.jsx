import React, { Component } from 'react';
import createSagaMiddleware from 'redux-saga';
import { combineReducers, createStore, compose, applyMiddleware } from 'redux';
import Application from 'terra-clinical-application';
import { Provider } from 'react-redux';
import './App.css';
import Popup from './Popup/PopupContainer';


const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers(Object.assign({}, null));

/** @const redux store */
const store = createStore(rootReducer, compose(applyMiddleware(sagaMiddleware)));

[].map(saga => (sagaMiddleware.run(saga)));

/* eslint-disable-next-line react/prefer-stateless-function */
class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Application>
          <Popup />
        </Application>
      </Provider>
    );
  }
}

export default App;
