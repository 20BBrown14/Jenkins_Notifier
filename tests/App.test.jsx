import React from 'react';
import App from '../src/App';

describe('App', () => {
  it('should render a default view', () => {
    const testApp = (shallow(<App />));
    expect(testApp).toMatchSnapshot();
  });
});
