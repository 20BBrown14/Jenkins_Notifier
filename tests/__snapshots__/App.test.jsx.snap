// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a default view 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Application>
    <Connect(PopupContainer) />
  </Application>
</Provider>
`;
